### PR TITLE
Upgrade Java to Beta Client Libraries

### PR DESCRIPTION
**Pull request by lindawoo-plaid**
Updates the Java backend from the plaid-java 8.1.0 Client API Libraries to the plaid-java 9.0.0 Beta-1 version

### CODE REVIEW

File: java/src/main/java/com/plaid/quickstart/QuickstartApplication.java [hunk 1]
```diff
@@ -52,40 +57,43 @@ public void initialize(final Bootstrap<QuickstartConfiguration> bootstrap) {
...
     String redirectUri = null;
     if (configuration.getPlaidRedirectUri() != null && configuration.getPlaidRedirectUri().length() > 0) {
       redirectUri = configuration.getPlaidRedirectUri();
     }
+
+    HashMap<String, String> apiKeys = new HashMap<String, String>();
+    apiKeys.put("clientId", plaidClientId);
+    apiKeys.put("secret", plaidSecret);
+    apiKeys.put("plaidVersion", "2020-09-14");
+    apiClient = new ApiClient(apiKeys);
```
Comment (stephenjayakar) [inline java/src/main/java/com/plaid/quickstart/QuickstartApplication.java:92]: 🎊

File: java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java [hunk 2]
```diff
@@ -1,60 +1,47 @@
...
-          ErrorResponse errorResponse = plaidClient.parseError(response);
-          Thread.sleep(3000);
-        } catch (InterruptedException e) {
-          throw Throwables.propagate(e);
-        }
-      }
-    }
-
-    return response.body();
+    LocalDate startDate = LocalDate.now().minusDays(365 * 2);
```
Comment (stephenjayakar) [inline java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java:31]: So initially there was a `retry` here, and it looks like there isn't in the new revision.
Does this fail sometime? Is the retry necessary?

File: ruby/Gemfile.lock [hunk 3]
```diff
@@ -34,4 +34,4 @@ DEPENDENCIES
   sinatra
 
 BUNDLED WITH
-   1.17.2
```
Comment (stephenjayakar) [inline ruby/Gemfile.lock:37]: This probably shouldn't be here in the java PR

**Review by stephenjayakar — COMMENTED** [review]
Great work :). Left some comments

File: ruby/Gemfile.lock (hunk 3 above)
Comment (lindawoo-plaid) [reply ruby/Gemfile.lock]: deleted

File: java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java (hunk 2 above)
Comment (lindawoo-plaid) [reply java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java:35]: I don't THINK the retry is necessary, but I added it back in.  (I didn't do it before cause it involved error handling, and I was gonna come back to that...)

File: java/src/main/java/com/plaid/quickstart/QuickstartApplication.java [hunk 4]
```diff
@@ -35,6 +36,10 @@
   // persistent data store.
   public static String paymentId;
 
+  private PlaidApi plaidClient;
+  private ApiClient apiClient;
+  public String env;
```
Comment (skylarmb) [inline java/src/main/java/com/plaid/quickstart/QuickstartApplication.java:41]: could this be more specific? env can refer to a lot of things, maybe this should be `plaidEnv`?

File: java/src/main/java/com/plaid/quickstart/resources/PaymentInitiationResource.java [hunk 5]
```diff
@@ -26,35 +26,40 @@
...
+      plaidClient
+      .paymentInitiationPaymentGet(payRequest)
+      .execute();
     if (!paymentGetResponse.isSuccessful()) {
       try {
-        ErrorResponse errorResponse = plaidClient.parseError(paymentGetResponse);
-        LOG.error("error: " + errorResponse);
+        Gson gson = new Gson();
+          Error errorResponse = gson.fromJson(paymentGetResponse.errorBody().string(), Error.class);
+          System.out.println(errorResponse);
```
Comment (skylarmb) [inline java/src/main/java/com/plaid/quickstart/resources/PaymentInitiationResource.java:50]: is there a reason we swapped `LOG.error` for `System.out.println`?

Comment (lindawoo-plaid) [reply java/src/main/java/com/plaid/quickstart/resources/PaymentInitiationResource.java:50]: nope. changed back

File: java/src/main/java/com/plaid/quickstart/QuickstartApplication.java (hunk 4 above)
Comment (lindawoo-plaid) [reply java/src/main/java/com/plaid/quickstart/QuickstartApplication.java]: changed to plaidEnv

File: .gitignore [hunk 6]
```diff
@@ -53,4 +53,10 @@ docker-compose.local*.yml
...
 java/.project
-java/.settings/
\ No newline at end of file
+java/.settings/
+
+#vs code
+.vscode
+
+#python
+python/.prettierrc
```
Comment (skylarmb) [inline .gitignore:62]: prettier does not work on python, so why is there a .prettierrc?

**Review by skylarmb — APPROVED** [review]

**Review by skylarmb — DISMISSED** [review]

File: java/.gitignore [hunk 7]
```diff
@@ -7,3 +7,8 @@ release.properties
 dependency-reduced-pom.xml
 buildNumber.properties
 .mvn/timing.properties
```
Comment (skylarmb) [inline java/.gitignore:9]:
```suggestion
.mvn
```

File: java/.gitignore [hunk 8]
```diff
@@ -7,3 +7,8 @@ release.properties
...
+
+#java
+.classpath
+.project
+.settings/
```
Comment (skylarmb) [inline java/.gitignore:14]:
```suggestion
.settings
```

**Review by skylarmb — CHANGES_REQUESTED** [review]

**Review by skylarmb — APPROVED** [review]

**Pull request merged**